packages/astro/src/@types/astro.ts

```diff
@@ -1714,6 +1715,7 @@ export interface AstroAdapterFeatures {
 export interface AstroSettings {
 	config: AstroConfig;
 	adapter: AstroAdapter | undefined;
+	preferences: AstroPreferences;
```

Comment:
It is incredibly funny to have config and preferences on a settings object, but I have no idea how we could make it clearer. (At least we're consistent, so it's not too bad!)